LinkedBug/LinkedBug.cs

```diff
@@ -0,0 +1,198 @@
+﻿namespace DataStructure;
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+/// <summary>
+/// A bag (multiset) data structure implemented using a linked list.
+/// </summary>
+public class LinkedBag<T> : ICollection<T>
+{
+    /// <summary>
+    /// A node in the linked list.
+    /// </summary>
+    private class Node
+    {
+        /// <summary>
+        /// Gets or sets the data stored in the node.
+        /// </summary>
+        public T Data { get; set; }
+        /// <summary>
+        /// Gets or sets the next node in the linked list.
+        /// </summary>
+        public Node Next { get; set; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Node"/> class.
+        /// </summary>
+        public Node(T data)
+        {
+            Data = data;
+            Next = null;
+        }
+    }
+
+    /// <summary>
+    /// The head of the linked list.
+    /// </summary>
+    private Node _head;
+    /// <summary>
+    /// The tail of the linked list.
+    /// </summary>
+    private Node _tail;
+    /// <summary>
+    /// The number of elements in the bag.
+    /// </summary>
+    private int _count;
+
+    /// <summary>
+    /// Initializes a new instance of the class.
+    /// </summary>
+    public LinkedBag()
+    {
+        _head = null;
+        _tail = null;
+        _count = 0;
+    }
+
+    /// <summary>
+    /// Gets the number of elements contained in the LinkedBag.
+    /// </summary>
+    public int Count => _count;
+
+    /// <summary>
+    /// Gets a value indicating whether the LinkedBag is read-only.
+    /// </summary>
+    public bool IsReadOnly => false;
+
+    /// <summary>
+    /// Adds an item to the LinkedBag
+    /// </summary>
+    public void Add(T item)
+    {
+        Node newNode = new Node(item);
```

Comment:
```suggestion
        Node newNode = new (item);
```
Не пишите один тип в одном объявлении дважды